### PR TITLE
translog-s3 plugin skeleton

### PR DESCRIPTION
**Pull request by satyajitg28**
### Description
translog s3 plugin skeleton, which will eventually be enhanced to add logic related to remote translog upload to s3.
 
### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [X] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugins/translog-s3/build.gradle [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
```
Comment (sachinpkale) [inline plugins/translog-s3/build.gradle:13]: This license text block is not required for new files. Only the first block is required.

File: plugins/translog-s3/build.gradle [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+opensearchplugin {
+  description 'The S3 translog plugin adds S3 translogs'
```
Comment (sachinpkale) [inline plugins/translog-s3/build.gradle:32]: `adds S3 translogs` doesn't seem right. Please re-word.

**Review by sachinpkale — CHANGES_REQUESTED** [review]
Please add a description to the PR. Even for skeleton PR, mention what it is trying to do and what are the next steps. Also, write commit message in the imperative mood

File: plugins/translog-s3/build.gradle (hunk 2 above)
Comment (satyajitg28) [reply plugins/translog-s3/build.gradle]: will do.

File: plugins/translog-s3/build.gradle (hunk 1 above)
Comment (satyajitg28) [reply plugins/translog-s3/build.gradle:13]: took the one under `repository-s3`

Comment (sachinpkale) [reply plugins/translog-s3/build.gradle:13]: `repository-s3` was part of Elasticsearch that is why the text says `Licensed to Elasticsearch`. For any new file in OpenSearch, this is not required.

**Review by Bukhtawar — COMMENTED** [review]
Please fix the license file to only include 
```
/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 */
```

**Comment by satyajitg28** [comment]
The following test passed on local:
```
./gradlew ':qa:remote-clusters:integTest' --tests "org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks"
```

**Review by sachinpkale — APPROVED** [review]

**Review by Bukhtawar — COMMENTED** [review]
Do we so commonalities with repository-s3 plugin that we might reuse and fold into?

**Comment by satyajitg28** [comment]
We have decided to leverage `repository-s3` for remote translog

**Pull request closed without merging**